pkg/deploy/lattice/listener_synthesizer.go

```diff
@@ -65,6 +66,7 @@ func (l *listenerSynthesizer) Synthesize(ctx context.Context) error {
 	latticeListenersAsModel, err := l.getLatticeListenersAsModels(ctx)
 	if err != nil {
 		listenerErr = errors.Join(listenerErr, err)
+		return listenerErr
```

Comment:
We already checked listenerErr not nil above. Should be plain err != nil return err, without join